docs/content/guides/dagster/branch-depl-with-snowflake.mdx

```diff
@@ -0,0 +1,348 @@
+
+We’ll first need to define a job that clones our `PRODUCTION` database for each branch deployment. Each clone will be named `PRODUCTION_CLONE_<ID>` with `<ID>` representing the pull request ID, ensuring each branch deployment has a unique clone. This job will drop a database clone if it exists and then reclone from production, ensuring each redeployment has a fresh clone of `PRODUCTION`:
+
+```python file=/guides/dagster/local_to_production/branch_depl_tutorial/clone_and_drop_db.py startafter=start_clone_db endbefore=end_clone_db
+@op(required_resource_keys={"snowflake"})
+def drop_database_clone(context):
+    context.resources.snowflake.execute_query(
+        f"DROP DATABASE IF EXISTS PRODUCTION_CLONE_{os.environ['DAGSTER_CLOUD_PULL_REQUEST_ID']}"
+    )
+
+
+@op(required_resource_keys={"snowflake"}, ins={"start": In(Nothing)})
+def clone_production_database(context):
+    context.resources.snowflake.execute_query(
+        f"CREATE DATABASE PRODUCTION_CLONE_{os.environ['DAGSTER_CLOUD_PULL_REQUEST_ID']} CLONE \"PRODUCTION\""
+    )
+
+
+@graph
+def clone_prod():
+    clone_production_database(start=drop_database_clone())
+```
+
+We’ve defined `drop_database_clone` and `clone_production_database` to utilize the [Snowflake resource](\_apidocs/libraries/dagster-snowflake#dagster_snowflake.snowflake_resource). The Snowflake resource will use the same configuration as the Snowflake I/O manager to generate a connection to Snowflake. However, while our I/O manager writes outputs to Snowflake, the Snowflake resource executes queries against Snowflake.
+
+We can then add the `clone_prod` job to our repository, configuring it with the resources corresponding to the current environment. We can modify the resource mapping by environment in our repository as follows:
+
+```python file=/guides/dagster/local_to_production/branch_depl_tutorial/repository_v2.py startafter=start_resources endbefore=end_resources
+resource_defs = {
+    "branch": {
+        "snowflake_io_manager": snowflake_io_manager.configured(
+            {
+                **snowflake_config,
+                "database": f"PRODUCTION_CLONE_{os.getenv('DAGSTER_CLOUD_PULL_REQUEST_ID')}",
+            }
+        ),
+        "snowflake": snowflake_resource.configured(
+            {
+                **snowflake_config,
+                "database": f"PRODUCTION_CLONE_{os.getenv('DAGSTER_CLOUD_PULL_REQUEST_ID')}",
+            }
+        ),
+    },
+    "production": {
+        "snowflake_io_manager": snowflake_io_manager.configured(
+            {
+                **snowflake_config,
+                "database": "PRODUCTION",
+            }
+        ),
+        "snowflake": snowflake_resource.configured({**snowflake_config, "database": "PRODUCTION"}),
+    },
+}
+```
+
+Then, we can add the `clone_prod` job to our repository:
```

Comment:
do you think people will be nervous about adding the `clone_prod` job to their repo with production credentials? Like I'm assuming this means that the `clone_prod` will show up as a runnable job in production dagit. It would fail since the `DAGSTER_CLOUD_PULL_REQUEST_ID` env var wouldn't exist, but it still might make people nervous to have it hanging around. not sure if there's anything we can do about that though

Reply:
hmm... yeah the job will fail if it attempts to run on production because the env var won't exist. We can update the job to automatically fail if the deployment isn't a branch deployment

Reply:
maybe this gets too complicated but can we optionally add the job to the repo if we're in the branch deployment environment? like

```
@repository 
def repo 

     all_defs = [
         with_resources([items, comments, stories], resource_defs=resource_defs[get_current_env()]),
    ]
    if get_current_env() == "branch":
         all_defs.append(clone_prod.to_job(resource_defs=resource_defs[get_current_env()]))

    return all_defs
```

I think this was pitched in one of the meetings where we talked about how to handle providing config based on environment. I can't remember if we decided we weren't going to recommend this approach though

Reply:
I updated this to conditionally add the branch deployment jobs